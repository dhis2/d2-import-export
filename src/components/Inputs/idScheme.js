import i18n from '@dhis2/d2-i18n'
import React from 'react'
import {
    RadioGroupContainer,
    RadioGroupLabel,
} from '../FinalFormComponents/RadioGroup'
import { Select } from '../FinalFormComponents/Select'

export const OPTION_UID = { value: 'UID', label: i18n.t('Uid') }
export const OPTION_CODE = { value: 'CODE', label: i18n.t('Code') }

export const ID_SCHEME_DEFAULT_OPTIONS = [OPTION_UID, OPTION_CODE]
export const ID_SCHEME_KEY = 'idScheme'
export const ID_SCHEME_DEFAULT_VALUE = OPTION_UID.value

const idSchemeLabel = i18n.t('Id scheme')
export const IdScheme = ({ options }) => (
    <Select name={ID_SCHEME_KEY} label={idSchemeLabel} options={options} />
)

IdScheme.propTypes = {
    options: Select.propTypes.options,
}

IdScheme.defaultProps = {
    options: ID_SCHEME_DEFAULT_OPTIONS,
}

export const IdSchemeLoading = () => (
    <RadioGroupContainer>
        <RadioGroupLabel>{idSchemeLabel}</RadioGroupLabel>
        {i18n.t('Loading organisation unit id scheme options...')}
    </RadioGroupContainer>
)
