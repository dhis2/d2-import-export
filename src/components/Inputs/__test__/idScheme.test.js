import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { IdScheme } from '../idScheme'

jest.mock('react-final-form', () => ({
    useField: jest.fn(() => ({
        input: {
            name: 'Name',
            value: '',
            onChange: jest.fn(),
        },
        meta: {
            touched: false,
            error: '',
        },
    })),
}))

describe('Input component - IdScheme', () => {
    it('should render correctly', () => {
        const file = shallow(<IdScheme />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
