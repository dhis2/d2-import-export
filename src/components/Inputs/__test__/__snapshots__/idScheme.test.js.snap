// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - IdScheme should render correctly 1`] = `
<Select
  data-test-id="input-id-scheme"
  label="Id scheme"
  name="idScheme"
  options={
    Array [
      Object {
        "label": "Uid",
        "value": "UID",
      },
      Object {
        "label": "Code",
        "value": "CODE",
      },
    ]
  }
/>
`;
